doc/guides/tips-tricks-and-troubleshooting.md

```diff
@@ -0,0 +1,74 @@
+# Tips, Tricks & Troubleshooting
```

Comment:
Called it "Tips & Troubleshooting for working with the `node` code"
Move the "Best practices" to a different doc